### PR TITLE
Feature/da app confirm contact details 1486

### PR DESCRIPTION
**Pull request by praveenrach**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -6,7 +6,7 @@ sonar.tests=src/test/
...
-sonar.exclusions=src/**/*.test.ts,**/__mocks__/**,src/**/*content.ts,src/**/components/*.ts,src/**/definition.ts,src/**/summary/util.ts,src/**/summary/util.ts,src/**/steps/constants.ts,src/**/willingness-to-attend-miam/miam-cost-exemptions.ts,src/**/steps/index.ts,src/**/steps/urls.ts,src/**/case/CaseApi.ts,src/**/app/controller/PostController.ts,src/**/checkanswers/ConfirmContactDetailsGetController.ts,src/**/respondent/task-list/utils.ts,src/**/respondent/respondentcaseSequence.ts,src/main/steps/common/summary/utils.ts
+sonar.exclusions=src/**/*.test.ts,**/__mocks__/**,src/**/*content.ts,src/**/components/*.ts,src/**/definition.ts,src/**/summary/util.ts,src/**/summary/util.ts,src/**/steps/constants.ts,src/**/willingness-to-attend-miam/miam-cost-exemptions.ts,src/**/steps/index.ts,src/**/steps/urls.ts,src/**/case/CaseApi.ts,src/**/app/controller/PostController.ts,src/**/checkanswers/ConfirmContactDetailsGetController.ts,src/**/respondent/task-list/utils.ts,src/**/respondent/respondentcaseSequence.ts,src/main/steps/common/summary/utils.ts,src/**/checkanswers/RespondentConfirmContactDetailsGetController.ts,src/**/checkanswers/ApplicantConfirmContactDetailsGetController.ts
```
Comment (cshiwarale) [inline sonar-project.properties:9]: why we need to add these controllers to sonar exclusions?

File: src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { GetController } from '../../../../app/controller/GetController';
+@autobind
+export default class ApplicantConfirmContactDetailsGetController extends GetController {
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    const redirect = false;
+
+    if (req.session?.user) {
+      res.locals.isLoggedIn = true;
+      req.locals.api = getCaseApi(req.session.user, req.locals.logger);
+      if (!req.session.userCase) {
```
Comment (cshiwarale) [inline src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts:16]: Please remove this commented code and the 'if' condition, as we have commented everything inside if loop

File: src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    // if(!req.session.userCase.applicant1DateOfBirth || req.session.userCase.applicant1DateOfBirth.day === '' || req.session.userCase.applicant1DateOfBirth.month === '' || req.session.userCase.applicant1DateOfBirth.year === ''){
+    //   req.session.userCase.applicant1PlaceOfBirthText = '<span class="govuk-error-message">Complete this section</span>';
+    // } else {
+    //   req.session.userCase.applicant1DateOfBirthText = JSON.stringify(req.session.userCase.applicant1DateOfBirth);
+    // }
+
+    req.session.userCase.applicant1Address1 = 'Flat 100';
+    req.session.userCase.applicant1Address2 = 'Plashet Grove';
+    req.session.userCase.applicant1AddressTown = 'London';
+    req.session.userCase.applicant1PhoneNumber = '';
```
Comment (cshiwarale) [inline src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts:65]: double initialisation of same attribute found

File: src/main/steps/applicant/task-list/utils.ts [hunk 4]
```diff
@@ -8,15 +8,25 @@ export const getKeepYourDetailsPrivateStatus = (userCase: CaseWithId): SectionSt
   if (userCase?.detailsKnown || userCase?.startAlternative) {
     return SectionStatus.IN_PROGRESS;
   }
-  return SectionStatus.NOT_STARTED;
+  return SectionStatus.TO_DO;
 };
 
 export const getConfirmOrEditYourContactDetails = (userCase: CaseWithId): SectionStatus => {
-  if (userCase?.confirmcontactdetails) {
+  if (userCase?.applicant1FullName && userCase?.applicant1DateOfBirth && userCase?.applicant1PlaceOfBirth) {
```
Comment (cshiwarale) [inline src/main/steps/applicant/task-list/utils.ts:15]: why are we only considering 3 attributes for making decision for completion of page? Do we require to consider other fields as well?

File: src/main/steps/common/confirm-contact-details/checkanswers/ConfirmContactDetailsGetController.ts [hunk 5]
```diff
@@ -38,8 +38,16 @@ export default class ConfirmContactDetailsGetController extends GetController {
     req.session.userCase.applicant1Address1 = 'Flat 100';
     req.session.userCase.applicant1Address2 = 'Plashet Grove';
     req.session.userCase.applicant1AddressTown = 'London';
-    req.session.userCase.applicant1PhoneNumber = '1234567890';
-    req.session.userCase.applicant1EmailAddress = 'test@gmail.com';
+    req.session.userCase.applicant1PhoneNumber = '';
```
Comment (cshiwarale) [inline src/main/steps/common/confirm-contact-details/checkanswers/ConfirmContactDetailsGetController.ts:41]: why we initialising this field to blank? can this field be always intended to be blank?

File: src/main/steps/respondent/task-list/utils.ts [hunk 6]
```diff
@@ -12,13 +12,13 @@ export const getKeepYourDetailsPrivateStatus = (userCase: CaseWithId): SectionSt
 };
 
 export const getConfirmOrEditYourContactDetails = (userCase: CaseWithId): SectionStatus => {
-  if (userCase?.confirmcontactdetails) {
+  if (userCase?.applicant1FullName && userCase?.applicant1DateOfBirth && userCase?.applicant1PlaceOfBirth) {
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/respondent/task-list/utils.ts:15]: As discussed, for this jira we considered first three fields only for page completion. 
Address & Address History, Email & Phone fields have separate tickets.

File: src/main/steps/applicant/task-list/utils.ts (hunk 4 above)
Comment (praveenrach) [reply src/main/steps/applicant/task-list/utils.ts:15]: As discussed, for this jira we considered first three fields only for page completion.
Address & Address History, Email & Phone fields have separate tickets.

File: src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts (hunk 2 above)
Comment (praveenrach) [reply src/main/steps/applicant/confirm-contact-details/checkanswers/ApplicantConfirmContactDetailsGetController.ts:16]: agreed, removed unwanted commented code

File: src/main/steps/respondent/confirm-contact-details/checkanswers/RespondentConfirmContactDetailsGetController.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+@autobind
+export default class RespondentConfirmContactDetailsGetController extends GetController {
+  public async get(req: AppRequest, res: Response): Promise<void> {
+    const redirect = false;
+
+    if (req.session?.user) {
+      res.locals.isLoggedIn = true;
+      req.locals.api = getCaseApi(req.session.user, req.locals.logger);
+      if (!req.session.userCase) {
+        //This language preference will be used while creating a case
```
[the remark this answers is not in the document]
Comment (praveenrach) [reply src/main/steps/respondent/confirm-contact-details/checkanswers/RespondentConfirmContactDetailsGetController.ts:17]: agreed, will remove unwanted/ commented code

File: sonar-project.properties (hunk 1 above)
Comment (praveenrach) [reply sonar-project.properties]: removed from the exclusions, implemented the unit tests for the same.

**Pull request merged**